core/server/common/src/main/java/alluxio/master/BackupManager.java

```diff
@@ -91,7 +99,15 @@ public void initFromBackup(InputStream is) throws IOException {
       while ((entry = reader.readEntry()) != null) {
         String masterName = JournalEntryAssociation.getMasterForEntry(entry);
         Master master = mastersByName.get(masterName);
-        master.processJournalEntry(entry);
+        try {
+          master.processJournalEntry(entry);
+        } catch (Throwable t) {
+          JournalUtils.handleJournalReplayFailure(
+              LOG, t, "Failed to process journal entry %s when init from backup", entry);
+          if (!mTolerantCorruption) {
```

Comment:
Can we move the throw into `handleJournalReplayFailure`? Wrapping the exception in a runtime exception is fine since we would only throw during tests.